### ISSUE TITLE
FB9987565: SwiftUI onDrag should support drag & drop onto the Trash icon in the Dock to delete

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#300]
- **Date:** 2022-04-19
- **Resolution:** Open
- **Area:** SwiftUI Framework
- **OS:** macOS 12.3
- **Type:** Incorrect/Unexpected Behavior

## Description

If you add `.onDrag()` to a view and then drag and drop the view onto the Trash icon in the Dock, the Trash icon accepts the drop, but nothing happens. I have received multiple complains from users that this behavior does not work. The immediate fix would be to not allow dropping on the Trash as that would let the user know it’s an unsupported action.

The ideal fix would be to support a `delegate` parameter in `.onDrag()`, similar to what `.onDrop()` has, which would let you decide what `NSDragOperation`’s the view supports and receive the operation the user chose.

In short, SwiftUI needs something similar to a `NSDraggingSource`: https://developer.apple.com/documentation/appkit/nsdraggingsource
